zcash_client_backend/proto/service.proto

```diff
@@ -34,6 +34,10 @@ message TxFilter {
 // RawTransaction contains the complete transaction data. It also optionally includes 
 // the block height in which the transaction was included, or, when returned
 // by GetMempoolStream(), the latest block height.
+//
+// FIXME: the documentation here about mempool status contradicts the documentation
+// for the `height` field (which I hope is correct, and this doesn't give the latest
+// block height for mempool transactions.)
```

Comment:
It turns out that the `GetTransaction` method added to `lightwalletd` is horribly broken, because it does not correctly propagate the output of the `getrawtransaction` JSON-RPC method. `getrawtransaction` returns:
- A non-negative height for transactions mined in blocks in the current main chain.
- `-1` for transactions mined in blocks that are not in the main chain (i.e. conflicted).
- Omitted for unmined transactions in the mempool.

This is further muddied by the fact that the Go parser is parsing the height field as an `int`, which is machine-dependent (and thus could differ from `zcashd`), and also I don't know how it handles omitted values (because it isn't to my knowledge a nullable type). So the unmined transactions might be getting their height fields coerced to some default (0?) - this needs further investigation.

`RawTransaction` was then further misused by the addition of the `GetMempoolStream` RPC. Transactions in the mempool should **never** be associated with the latest block height; if any height is associated with them, it should be the "mempool height" which is `latest_height + 1` (which is the height at which the consensus rules are applied to the mempool transactions). So that RPC is broken as well.

This is going to be Not Fun™️ to fix.

Reply:
>This is going to be Not Fun™️ to fix. 


I've pinged wallet developers that might affected by this to bring up awareness.

Reply:
We have discovered additional underlying issues: https://github.com/ZcashFoundation/zebra/issues/8744

Reply:
Lightwalletd issues documented here: https://github.com/zcash/lightwalletd/issues/493

Reply:
Filed as https://github.com/zcash/librustzcash/issues/1484